### PR TITLE
add tests

### PR DESCRIPTION
**Pull request by qreqit**

### CODE REVIEW

File: src/main/java/mate/academy/springbootwebgreqit/controller/ShoppingCartController.java [hunk 1]
```diff
@@ -21,25 +22,25 @@ public class ShoppingCartController {
     private final ShoppingCartService shoppingCartService;
 
     @GetMapping
-    public ShoppingCartDto getShoppingCartForCurrentUser(@RequestParam Long userId) {
-        return shoppingCartService.getShoppingCartForCurrentUser(userId);
+    public ShoppingCartDto getShoppingCartForCurrentUser(Authentication authentication) {
+        return shoppingCartService.getShoppingCartForCurrentUser(authentication);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootwebgreqit/controller/ShoppingCartController.java:26]: It's better to create separate method that return Long userId from Authentication

File: src/test/java/mate/academy/springbootwebgreqit/controller/ShoppingCartControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+    @BeforeAll
+    static void beforeAll(@Autowired WebApplicationContext webApplicationContext) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(webApplicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @Sql(scripts = {"/data-sql/create-books.sql",
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/springbootwebgreqit/controller/ShoppingCartControllerTest.java:43]: replace sql scripts under the test class

File: src/test/java/mate/academy/springbootwebgreqit/service/ShoppingCartServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+import mate.academy.springbootwebgreqit.repository.UserRepository;
+import mate.academy.springbootwebgreqit.service.impl.ShoppingCartServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.security.core.Authentication;
+
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/springbootwebgreqit/service/ShoppingCartServiceTest.java:24]: looks like checkstyle doesn't work well 
check it please

File: src/test/java/mate/academy/springbootwebgreqit/service/ShoppingCartServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+        cartItem.setBook(book);
+        cartItem.setQuantity(2);
+        cartItem.setShoppingCart(shoppingCart);
+
+        authentication = mock(Authentication.class);
+    }
+
+    @Test
+    void getShoppingCartForCurrentUser_ShouldReturnShoppingCartDto() {
+        user.setShoppingCart(shoppingCart);
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/springbootwebgreqit/service/ShoppingCartServiceTest.java:87]: need to fix main logic first and then refactor tests

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
first fix main logic
I've reviewed you code in main branch - there are a lot of mistake that should be fixed

File: src/main/java/mate/academy/springbootwebgreqit/dto/shoppingcart/RequestUpdateQuantityDto.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package mate.academy.springbootwebgreqit.dto.shoppingcart;
+
+import lombok.Data;
+
+@Data
+public class RequestUpdateQuantityDto {
+    private int quantity;
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootwebgreqit/dto/shoppingcart/RequestUpdateQuantityDto.java:7]: add validation

File: src/main/java/mate/academy/springbootwebgreqit/dto/shoppingcart/RequestUpdateQuantityDto.java [hunk 6]
```diff
@@ -0,0 +1,8 @@
+package mate.academy.springbootwebgreqit.dto.shoppingcart;
+
+import lombok.Data;
+
+@Data
+public class RequestUpdateQuantityDto {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootwebgreqit/dto/shoppingcart/RequestUpdateQuantityDto.java:6]:
```suggestion
public class UpdateCartItemDto {
```

File: src/main/java/mate/academy/springbootwebgreqit/service/impl/ShoppingCartServiceImpl.java [hunk 7]
```diff
@@ -31,16 +34,20 @@ public class ShoppingCartServiceImpl implements ShoppingCartService {
 
     @Transactional
     @Override
-    public ShoppingCartDto getShoppingCartForCurrentUser(Long userId) {
-        User user = userRepository.findById(userId)
-                .orElseThrow(() -> new IllegalArgumentException("User not found"));
-        ShoppingCart shoppingCart = user.getShoppingCart();
+    public ShoppingCartDto getShoppingCartForCurrentUser(Authentication authentication,
+                                                         Long shoppingCartId) {
+        User authUser = userRepository.findByEmail(authentication.getName())
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootwebgreqit/service/impl/ShoppingCartServiceImpl.java:39]:
```suggestion
        User user = (User) authentication.getPrincipal();
        ShoppingCart shoppingCart = shoppingCartRepository.findById(user.getUserId())....
```

File: src/main/java/mate/academy/springbootwebgreqit/service/impl/ShoppingCartServiceImpl.java [hunk 8]
```diff
@@ -31,16 +34,20 @@ public class ShoppingCartServiceImpl implements ShoppingCartService {
...
+        ShoppingCart shoppingCart = shoppingCartRepository.findById(shoppingCartId)
+                .orElseThrow(() -> new EntityNotFoundException("Shopping cart not "
+                        + "found for user with id: " + shoppingCartId));
         if (shoppingCart == null) {
-            throw new EntityNotFoundException("Shopping cart not found for user with ID "
-                    + userId);
+            throw new EntityNotFoundException("Shopping cart not found for user with email: "
+                    + authUser.getEmail());
         }
         Hibernate.initialize(shoppingCart.getCartItems());
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootwebgreqit/service/impl/ShoppingCartServiceImpl.java:49]: Why do you need these all Hibernate.initialize?

File: src/main/java/mate/academy/springbootwebgreqit/service/impl/ShoppingCartServiceImpl.java [hunk 9]
```diff
@@ -52,8 +59,11 @@ public ShoppingCartDto getShoppingCartForCurrentUser(Long userId) {
     @Override
     public ShoppingCartDto addBookToShoppingCart(
             CartItemRequestDto cartItemDto,
-            Long userId) {
-        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId)
+            Authentication authentication) {
+        User user = userRepository.findByEmail(authentication.getName())
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootwebgreqit/service/impl/ShoppingCartServiceImpl.java:63]: just get User from Principle (comment above)

File: src/test/java/mate/academy/springbootwebgreqit/controller/ShoppingCartControllerTest.java (hunk 2 above)
Comment (Elena-Bruyako) [reply src/test/java/mate/academy/springbootwebgreqit/controller/ShoppingCartControllerTest.java:46]: fix comment

File: src/test/java/mate/academy/springbootwebgreqit/service/ShoppingCartServiceTest.java (hunk 3 above)
Comment (Elena-Bruyako) [reply src/test/java/mate/academy/springbootwebgreqit/service/ShoppingCartServiceTest.java:25]: Have you checked?

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/mate/academy/springbootwebgreqit/service/impl/ShoppingCartServiceImpl.java (hunk 8 above)
Comment (qreqit) [reply src/main/java/mate/academy/springbootwebgreqit/service/impl/ShoppingCartServiceImpl.java:49]: iused it to initialize data, but now i don't need it, cause i use @Transactional

File: src/main/java/mate/academy/springbootwebgreqit/dto/shoppingcart/UpdateCartItemDto.java [hunk 10]
```diff
@@ -0,0 +1,10 @@
+package mate.academy.springbootwebgreqit.dto.shoppingcart;
+
+import jakarta.validation.constraints.NotNull;
+import lombok.Data;
+
+@Data
+public class UpdateCartItemDto {
+    @NotNull
```
Comment (okuzan) [inline src/main/java/mate/academy/springbootwebgreqit/dto/shoppingcart/UpdateCartItemDto.java:8]: primitive cannot be null
maybe you meant @Positive?

File: src/test/resources/data-sql/create-cart-item.sql [hunk 11]
```diff
@@ -0,0 +1 @@
+INSERT INTO cart_items (id, shopping_cart_id, book_id, quantity) VALUES (1, 1, 1, 2);
```
Comment (okuzan) [inline src/test/resources/data-sql/create-cart-item.sql:1]: image
minor issue but still, always review your files and make sure there's one empty line at the end

File: src/test/java/mate/academy/springbootwebgreqit/service/ShoppingCartServiceTest.java [hunk 12]
```diff
@@ -0,0 +1,185 @@
...
+    void createShoppingCart_ShouldCreateAndReturnShoppingCart() {
+        when(shoppingCartRepository.save(any(ShoppingCart.class))).thenReturn(shoppingCart);
+
+        ShoppingCart result = shoppingCartService.createShoppingCart(user);
+
+        assertNotNull(result);
+        assertEquals(user, result.getUser());
+        verify(shoppingCartRepository).save(result);
+    }
+}
```
Comment (okuzan) [inline src/test/java/mate/academy/springbootwebgreqit/service/ShoppingCartServiceTest.java:185]:
```suggestion
}

```

File: src/test/java/mate/academy/springbootwebgreqit/service/ShoppingCartServiceTest.java [hunk 13]
```diff
@@ -0,0 +1,185 @@
...
+    }
+
+    @Test
+void updateCartItemQuantity_ShouldUpdateQuantityAndReturnShoppingCartDto() {
+    int newQuantity = requestUpdateQuantityDto.getQuantity();
+    when(authentication.getPrincipal()).thenReturn(user);
+    when(userRepository.findByEmail(user.getEmail())).thenReturn(Optional.of(user));
+    when(cartItemRepository.findById(cartItem.getId())).thenReturn(Optional.of(cartItem));
+    when(shoppingCartRepository.findByUserId(user.getId())).thenReturn(Optional.of(shoppingCart));
+    when(shoppingCartMapper.toDto(shoppingCart)).thenReturn(shoppingCartDto);
```
Comment (okuzan) [inline src/test/java/mate/academy/springbootwebgreqit/service/ShoppingCartServiceTest.java:142]: fix tabulation

File: src/test/java/mate/academy/springbootwebgreqit/controller/ShoppingCartControllerTest.java [hunk 14]
```diff
@@ -0,0 +1,126 @@
...
+    static void beforeAll(@Autowired WebApplicationContext webApplicationContext) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(webApplicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+
+    @Test
+    @Sql(scripts = {"/data-sql/create-books.sql",
```
Comment (okuzan) [inline src/test/java/mate/academy/springbootwebgreqit/controller/ShoppingCartControllerTest.java:46]: you have the same setup in each test method, move to class level

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by ahoienko — APPROVED** [review]
Good job!

**Pull request merged**